From a015291dc94b5497ac1fd60a2e4cf84fe3c82cb5 Mon Sep 17 00:00:00 2001
From: Joseph Zikusooka <joseph@jambula.net>
Date: Sun, 3 Nov 2024 15:09:56 +0300
Subject: [PATCH 1/1]  Kea: new package - kea version 2.6.1

 Add Kea, an open-source Dynamic Host Configuration Protocol (DHCP)
 server developed by the Internet Systems Consortium (ISC).
 It's designed to provide dynamic IP address allocation and
 configuration for devices on a network. Kea is known for its
 modular architecture, allowing for easy extension and integration
 with other services.

Signed-off-by: Joseph Zikusooka <joseph@jambula.net>
---
 package/Config.in     |  1 +
 package/kea/Config.in | 16 ++++++++++++++++
 package/kea/kea.hash  |  3 +++
 package/kea/kea.mk    | 17 +++++++++++++++++
 4 files changed, 37 insertions(+)
 create mode 100644 package/kea/Config.in
 create mode 100644 package/kea/kea.hash
 create mode 100644 package/kea/kea.mk

diff --git a/package/Config.in b/package/Config.in
index 876b340a8a..4bfc14bdbf 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -2516,6 +2516,7 @@ endif
 	source "package/iw/Config.in"
 	source "package/iwd/Config.in"
 	source "package/janus-gateway/Config.in"
+	source "package/kea/Config.in"
 	source "package/keepalived/Config.in"
 	source "package/kismet/Config.in"
 	source "package/knock/Config.in"
diff --git a/package/kea/Config.in b/package/kea/Config.in
new file mode 100644
index 0000000000..508671a983
--- /dev/null
+++ b/package/kea/Config.in
@@ -0,0 +1,16 @@
+config BR2_PACKAGE_KEA
+
+	bool "kea"
+
+	select BR2_PACKAGE_BOOST
+	select BR2_PACKAGE_OPENSSL
+	select BR2_PACKAGE_LOG4CPLUS
+
+	help
+	  DHCP implementation from Internet Systems Consortium, Inc. that features fully
+	  functional DHCPv4, DHCPv6 and Dynamic DNS servers.
+	  Both DHCP servers fully support server discovery, address assignment, renewal,
+	  rebinding and release. The DHCPv6 server supports prefix delegation. Both
+	  servers support DNS Update mechanism, using stand-alone DDNS daemon
+
+          https://www.isc.org/kea
diff --git a/package/kea/kea.hash b/package/kea/kea.hash
new file mode 100644
index 0000000000..c362dadbb1
--- /dev/null
+++ b/package/kea/kea.hash
@@ -0,0 +1,3 @@
+# Locally computed - sha256
+sha256  d2ce14a91c2e248ad2876e29152d647bcc5e433bc68dafad0ee96ec166fcfad1  kea-2.6.1.tar.gz
+sha256  bb93e968a3c2f0487938d245c51b11edda048c1900cd3fdfd9897e48ce6b838f  COPYING
diff --git a/package/kea/kea.mk b/package/kea/kea.mk
new file mode 100644
index 0000000000..a8bdb9c4e0
--- /dev/null
+++ b/package/kea/kea.mk
@@ -0,0 +1,17 @@
+################################################################################
+#
+# kea
+#
+################################################################################
+
+KEA_VERSION = 2.6.1
+KEA_SOURCE = kea-$(KEA_VERSION).tar.gz
+KEA_SITE = https://dl.cloudsmith.io/public/isc/kea-2-6/raw/versions/$(KEA_VERSION)
+
+KEA_LICENSE = MPL 2.0
+KEA_LICENSE_FILES = COPYRIGHT
+
+KEA_INSTALL_STAGING = YES
+KEA_INSTALL_TARGET = YES
+
+$(eval $(autotools-package))
-- 
2.40.1

